src/instana/agent/host.py

```diff
@@ -346,13 +346,17 @@ def report_spans(self, payload: Dict[str, Any]) -> Optional[Response]:
         return
 
     def filter_spans(self, spans: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
-        from instana.util.traceutils import is_service_or_endpoint_ignored
+        from instana.util.traceutils import (
+            is_service_or_endpoint_ignored,
+            get_operation_specifier,
+        )
```

Comment:
Could you please move these imports to the top of the file?

Reply:
These imports cause circular imports on top. Could we keep it as is?

Reply:
If there are circle import errors, the code is doing something it is not supposed to do or was not well designed. I think you should try to solve them.

Reply:
Solved!